### ISSUE TITLE
老哥 这个在vue项目中怎么引入啊

### ISSUE THREAD

**Issue by xdyz** [534591395/smart_web_data_sdk#8]

**Comment by 534591395** [comment]
一个直接在html页面里引入build下的文件（打包好的），还有就是参考loader里的文件引入方式。

**Comment by xdyz** [comment]
用的脚手架3 没有html

**Comment by xdyz** [comment]
> 一个直接在html页面里引入build下的文件（打包好的），还有就是参考loader里的文件引入方式。

就是说 我reruire 进去 然后在插件里面new就可以了吗？

**Comment by xdyz** [comment]
> 一个直接在html页面里引入build下的文件（打包好的），还有就是参考loader里的文件引入方式。

老哥 这个具体的引入 是怎么样的啊 还是搞不懂

**Comment by 534591395** [comment]
import SMARTLib from '../src/main'

class LoaderSync {
  constructor() {
    window['smart'] = this;
  }
  init(token, config) {
    if (this['__loaded']) {
      return ;
    }
    this.instance = new SMARTLib(token, config);
    this.instance.init = this['init'];
    window['smart'] = this.instance;
  }
}

new LoaderSync();

类似这样

**Comment by xdyz** [comment]
> import SMARTLib from '../src/main'
> 
> class LoaderSync {
> constructor() {
> window['smart'] = this;
> }
> init(token, config) {
> if (this['__loaded']) {
> return ;
> }
> this.instance = new SMARTLib(token, config);
> this.instance.init = this['init'];
> window['smart'] = this.instance;
> }
> }
> 
> new LoaderSync();
> 
> 类似这样

我在我的vue 的index.html文件中引入了这个 也初始化了 但是好像跳转 按钮点击都没有发送请求啊 浏览器的回退前进可以 我用的vuerouter 配合elementui导航 的跳转就不起作用了 这个怎么回事

**Comment by 534591395** [comment]
开启了PV单页面监听了不？点击按钮加了埋点没？ 不行的话，你调试下源码看看？源码很简单的，都加了注释。

**Comment by xdyz** [comment]
> 开启了PV单页面监听了不？点击按钮加了埋点没？ 不行的话，你调试下源码看看？源码很简单的，都加了注释。

老哥 那就是说对于所有的按钮点击事件 我都需要去 加入   
document.getElementsByTagName('h1')[0].onclick = function(e) {
          smart.login(document.getElementById('userId').value);
        } 类似于这种方式 对吧

**Comment by 534591395** [comment]
是的。无埋点模块我没上传的，这个很简单，你可以自己实现下。这方面你也可以看下我另一个采集器的实现。  这个采集器主要还是代码埋点为主。

**Comment by xdyz** [comment]
明白了 我得对所有的点击事件进行smart.xxx({})进行 埋点 对吧

**Comment by xdyz** [comment]
> 是的。无埋点模块我没上传的，这个很简单，你可以自己实现下。这方面你也可以看下我另一个采集器的实现。 这个采集器主要还是代码埋点为主。

老哥 怎么结合你这个代码 实现一个无埋点的方式啊 我不知道怎么去修改你这个代码

**Comment by 534591395** [comment]
其实很简单的，比如全局监听click事件，然后拿到元素的信息，元素事先加些要采集的内容，再调用下track方法就好了。
